0017-chart-namespace-overrides/README.md

```diff
@@ -0,0 +1,440 @@
+
+## Proposal
+
+The proposed solution is to add a new `namespace` field under the Zarf Package configuration `metadata` and to allow a default to be set and allow this value to be exposed in Go templating under `charts`, `manifests`, and `actions`.  Charts and manifests would allow templating of their `namespace` and `releaseName` fields and actions would be templateable as designed in [ZEP-0021](./0021-zarf-values/README.md).
+
+### User Stories (Optional)
+
+#### Story 1
+
+**As** Jacquline **I want** to be able to set namespace values **so that** I can install the same package with different configurations in different namespaces.
+
+**Given** I have a Zarf Package created from the following
+```yaml
+kind: ZarfPackageConfig
+metadata:
+  name: example
+  ref: oci://my-registry/test:0.1.0
+  version: 0.1.0
+  namespace: example-namespace
+
+components:
+  - name: example-component
+    charts:
+      - name: example-chart
+        # option 1
+        namespace: "{{ .Package.metadata.namespace }}"
+        # option 2
+        namespace: "{{ .Release.Namespace }}"
+        # option 3
+        namespace: "{{ .Deploy.Namespace }}"
+        url: https://example.com/helm-chart
+    actions:
+      onDeploy:
+        after:
+          - wait:
+              cluster:
+                kind: Deployment
+                name: example
+                # option 1
+                namespace: "{{ .Package.metadata.namespace }}"
+                # option 2
+                namespace: "{{ .Release.Namespace }}"
+                # option 3
+                namespace: "{{ .Deploy.Namespace }}"
+                condition: "Available"
+```
+**When** I deploy that package with a `--namespace` flag like the below:
+```bash
+zarf package deploy oci://my-registry/test:0.1.0 --namespace new-namespace
+```
+**Or When** I deploy that package with a `zarf-config.yaml` like the below:
+```yaml
+package:
+  deploy:
+    namespace: new-namespace
+```
+**Then** Zarf will change the chart's release namespace to `new-namespace`
+**And** Zarf will change the action's cluster namespace to `new-namespace`
+
+### Risks and Mitigations
+
+We would need to be careful to document and clearly outline which fields are templateable and which are not - fields that control the contents of the Zarf package (i.e. `url`, `valuesFiles`, `name`, etc.) should not be templated since, if changed, they would refer to things that cannot be deployed in the airgap.  Only fields that control the deployment of the chart or manifest would be templateable and this would need to be clearly communicated to reduce confusion.  It may also be useful to outline areas beyond this proposal that Go templating should apply within a Zarf package definition to further reduce confusion.  We also should fail fast during package create if a template is found somewhere it is not allowed - this will allow a package creator to realize that they need to make a change before they attempt to deploy the package.
```

Comment:
Big :+1:, ideally when erroring out we should present that list of overridable fields to the user.